### PR TITLE
Feat agent reports

### LINKED ISSUE #4: [Agent] Reports for single arrangement

### PR DESCRIPTION
**Pull request by elkcert**
Closes #4, #5
Todo:
- [x] Fix {Prev, Next} page labels (how to bind through `Paginator`?)
- ~~Style ComboBox~~
- [ ] Prettify

### CODE REVIEW

File: YouTravel/Agent/ArrangementList.xaml.cs [hunk 1]
```diff
@@ -13,19 +13,9 @@
...
-
-		public ObservableCollection<Arrangement> Arrangements { get; } = new();
-		public ObservableCollection<Arrangement> ArrangementsCurrentPage { get; } = new();
-
-        private int _pageIndex = 0;
-        private int _pageCount = 0;
-        public int PageIndex { get { return _pageIndex; } set { _pageIndex = value; DoPropertyChanged(nameof(PageIndex)); SetPageNavButtonsEnabled(); } }
-		public int PageCount { get { return _pageCount; } set { _pageCount = value; DoPropertyChanged(nameof(PageCount)); SetPageNavButtonsEnabled(); } }
-        int pageSize = 2;
+        public Paginator<Arrangement> Paginator = new();
```
Comment (magley) [inline YouTravel/Agent/ArrangementList.xaml.cs]: To bind paginator pages, make this a property:
```cs
public Paginator<Arrangement> Paginator {get; set; } = new();
```
and change the binding path in the xaml:
```xml
<Label Content="{Binding Path=Paginator.PageIndex, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"/>
```
Note the `Paginator.PageIndex`.

File: YouTravel/Agent/ArrangementReport.xaml.cs [hunk 2]
```diff
@@ -1,44 +1,96 @@
...
+        {
+            LoadReservationsCurrentPage();
+        }
+
+        private void OnReservationCurrentPageCollectionChanged(object? sender, NotifyCollectionChangedEventArgs e)
+        {
+            ToggleNoPagesText();
+        }
+
+        private void ToggleNoPagesText()
```
Comment (magley) [inline YouTravel/Agent/ArrangementReport.xaml.cs:54]: Call this in the Loaded event/ctor, as well. If the arrangement doesn't have any reservations, then `CollectionChanged` doesn't get fired at all so both the table and "no reservations" are visible.

**Review by magley — CHANGES_REQUESTED** [review]
Monthly reports:

- [x] Why does January show 2 reservations, everything (at app startup, today) will be in June?
- [x] Move the combo box up
- [x] Instead of a combo box, use a label and 2 buttons for navigation between months (so we don't have to style yet another element + navigating through months in a combo box is unpleasant.

General:

- [x] Add more fields in the `Reservation` class, so that you don't have to add more columns later (username, for how many people did he book, a bool if he paid (the table column would be if he paid, the model would have a DateTime)

File: YouTravel/Agent/ArrangementList.xaml.cs (hunk 1 above)
Comment (elkcert) [reply YouTravel/Agent/ArrangementList.xaml.cs]: Forgot the getter and setter for the public field... https://github.com/magley/hci-travel-agent/pull/25/commits/46093d48ab5d542a1ab927acea09f7e63e43520a

File: YouTravel/Agent/ArrangementReport.xaml.cs (hunk 2 above)
Comment (elkcert) [reply YouTravel/Agent/ArrangementReport.xaml.cs:54]: https://github.com/magley/hci-travel-agent/pull/25/commits/045a098b2a80cf4ec3e6484dd5efef3b3b2362be

**Pull request closed without merging**